### ISSUE TITLE
Feature request: automatic refresh V2

### ISSUE THREAD

**Issue by nickion** [zanllp/sd-webui-infinite-image-browsing#138]
Great to see the automatic refresh added, but I wondered if there could be an alternative of triggering refresh on view activation. This would mirror the typical manual process of: 1) activate tab; 2) click refresh. The process is very fast operating locally with an SSD, so should work fine. There could be an optimisation/option of only refreshing the view if a new directory or new files in the most recent directory are detected.

**Comment by zanllp** [comment]
I didn't quite understand your point. Are you suggesting to refresh only the active tab each time?

**Comment by nickion** [comment]
Yes. I suspect that when activating a view, many users currently would first manually refresh the view to ensure it's up-to-date. If the plugin could do this automatically, it would ensure an accurate and up-to-date view of the files by default, with no user effort required.

Detailed Description:

The auto-refresh feature would trigger a refresh every time a tab becomes visible. This would keep the thumbnail view in sync with the file system, no matter how and when changes occurred.

To avoid unnecessary UI updates, the DOM should be updated only if there have actually been any changes. If the plugin uses a system such as the Watchdog library to detect file changes outside of A1111, it may already have an accurate view of the file system when a tab is activated, avoiding the need to actually scan the file system. However, to ensure 100% accuracy, a scan would be beneficial. If Watchdog is used currently or in the future, it could be an optional preference.

While the manual refresh operation is quick in my experience, and would be at least as efficient were it automatic, for some users there may be performance concerns. Hence, the feature should be optional.

UI Changes:

Auto-Refresh: Add as a toggle button (on/off) on the main interface for easy access. Additionally, consider a user preference to disable the feature permanently, removing the toggle button in that case.

Smart-Scan preference: If scanning the file system for changes is efficient on a user's system, the plugin could simply always scan the file system on a tab switch. However refresh could likely be skipped if the plugin is aware that there have been no image generations since the last tab refresh. This would optimise for a user who is switching between tabs for a while without generating images. Whether to smart-scan or always-scan could be a preference. Alternatively, as auto refresh could be disabled easily, the user could manually disable the feature temporarily if it was intrusive.

I wasn't fully clear myself how the auto refresh that was added recently behaves. If it is guaranteed that a tab view will be 100% accurate with the recent update, then this proposal is unnecessary, but otherwise, it should achieve the goal of complete view accuracy.

**Comment by zanllp** [comment]
Don't worry about the performance or update issues, the refresh mechanism is designed to be very simple and straightforward, yet still very fast and effective. I think it's already sufficient for now. 


This is the performance under 2000 files in a single folder (on an HDD). If someone needs higher performance than this, I think they should sponsor me to make it happen.


https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/25872019/cacd0d4e-8379-4ea3-94c8-b97490fd1542

**Comment by nickion** [comment]
Agreed. In my experience performance is fine too, but there could be someone storing files over a slow network share where performance is much slower. Being mindful of that and having a way to have a way to cater for those folks would be good design, just in case. 

I did check the latest version of the plugin just in case the issue has been solved and the need for this proposal has gone away, but if I generate a new image and switch to the browsing tab the image still does not show until I do a manual refresh, which is a nuisance. So that's is the motivation of proposal, which is hopefully clear now.
